# CellCosmo_UHT V1.2
Ultra-High-Throughput Single-Cell Transcriptomics Technology
* Propose
   * A collection of bioinformatics analysis pipelines to process single-cell sequencing data generated by Ultra-High-Throughput single-cell RNA products.
* Language
   * Python3(>=3.10.*)
   * R scripts
* Hardware/Software environment
   * x86-64 compatible processors.
   * require at least 40GB of RAM and 4 CPU cores.
   * centos 7.x 64-bit operating system (Linux kernel 3.10.0, compatible with higher software and hardware configurations).
# Installation
Installation tutorial manual [here](docs/install.md)

# Start Running CellCosmo_UHT
## 1. Build References  genomeDir for Homo sapiens or Mus musculus
Build References genomeDir tutorial manual [here](docs/Build_References_genomeDir.md)

## 2.Run pipeline
```bash
# Activate the Conda environment named "CellCosmo_UHT"
# This environment contains the required Python packages for the analysis
conda activate CellCosmo_UHT

# Set the path to the CellCosmo_UHT script directory
# Replace ${path} with the actual parent directory path
script_path=${path}/CellCosmo_UHT

# Set the path to the STAR genome index (GRCh38 reference genome)
STARindex=GRCh38_index/

work path:
0-data/test1_R1.fastq.gz
0-data/test1_R2.fastq.gz

# Execute the CellCosmo_UHT Python pipeline with specified parameters
python ${script_path}/CellCosmo_UHT.py \
--script_path ${script_path} \
--Rawdata_path ./0-data \
--SampleName 'test1' \
--CB3_Num '1-20' \
--STARindex /path/GRCh38_index \
--TopCells 3000 \
--STARsoloThreads 2 \
--SplitcodeNum 10 \
--nFastqs 2 \
--soloFeatures 'GeneFull_Ex50pAS Velocyto' \
--outRaw True \
--Summary summary.txt \
--Mapping mapping.txt \
--LogFile log.txt \
--splitCB "[0,88,8] [1,89,8] [2,90,8] [3,91,8] [4,92,8] [5,93,8] [6,94,8] [7,95,8]" \
--splitSample "A B C D E F G H"
```

# Support
The officially supported release binaries are available at: (http://www.10kgenomics.com/)
